homework/index.js

```diff
@@ -28,23 +28,56 @@
     });
     return elem;
   }
+  function renderRepoDetails(table, title, repoText, article, repoHref = '') {
```

Comment:
You have changed the purpose of this function. Its original purpose was to render the complete details of a repository. Now it renders only a table row of those details. I suggest you rename this function `addRow()` and reintroduce a function `renderRepoDetails()`. I give some further hint on that below.

This function for adding a row has become a bit complicated because of the first row that deviates from the other rows, in that it requires an `<a>` tag instead of simple text content. I would advise that you let the caller of this function deal with the exception for the first row and simplify this function to:

```js
function addRow(table, title, repoText) {
  const tr = createAndAppend('tr', table);
  createAndAppend('th', tr, {
    text: title,
  });
  createAndAppend('rd', tr, { text: repoText });
  return tr;
}
```

Notice that this function return the `tr` that was created so that the caller of this function can access this `tr`.

Next, all the code you had in the `.forEach()` inside function `main()` can now go to the original function `renderRepoDetails()` like this:

```js
function renderRepoDetails(repo, ul) {
  const li = createAndAppend('li', ul, { class: 'li-root' });
  const table = createAndAppend('table', li);
  const date = new Date(repo.updated_at);
  const firstRow = addRow(table, 'Repository:', '');
  createAndAppend('a', firstRow.lastChild, {
    href: repo.html_url,
    text: repo.name,
    target: '_blank',
  });
  addRow(table, 'Description:', repo.description, 'td');
  addRow(table, 'Forks:', repo.forks, 'td');
  addRow(table, 'Updated:', date.toLocaleDateString(), 'td');
}
```

As you can see, this function now deals with adding the `<a>` after creating the first row with an empty text content. In this way, the `addRow()` function can be kept simple: it no longer has to deal with the exception for the first row.

As you can see, we let this function add the

Reply:
My real problem is to create new and hoisting function. This is a really good example. Thank you. I am working on it. 👍